spmd/tensor/parallel/style.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates
+import torch
+from dataclasses import dataclass
+from typing import Callable, Union, Optional
+from spmd.tensor import (
+    DTensor,
+    Shard,
+    Replicate,
+    DeviceMesh,
+)
+
+
+@dataclass
+class ParallelStyle(object):
+    """
+    The parallel style user wants the module or submodule to be sharded.
+    Users can extend this class to build their own parallel style with customized input/output preparations.
+    """
+
+    _prepare_input: Optional[
+        Callable[
+            [Union[torch.Tensor, DTensor], Optional[DeviceMesh], Optional[int]],
+            DTensor,
+        ]
+    ]
+    _prepare_output: Optional[
+        Callable[
+            [DTensor, Optional[DeviceMesh], Optional[int]],
+            Union[torch.Tensor, DTensor],
+        ]
+    ]
+
+
+def shard_input_1d(
```

Comment:
Also per PyTorch coding standard, you need to add comment to this function with the spec of input/output.

Reply:
One example for this: https://github.com/pytorch/pytorch/blob/e2f0648750f2d0d0ac648728ce4c514db178cfa1/torch/distributed/_shard/sharded_tensor/api.py#L785.